lib/state/ui/reducer.ts

```diff
@@ -34,6 +34,21 @@ const visiblePanes: A.Reducer<string[]> = (
   return state;
 };
 
+const showNoteInfo: A.Reducer<boolean> = (state = false, action) => {
+  if ('NOTE_INFO_TOGGLE' === action.type) {
+    return action.show;
```

Comment:
considering that our only user-initiated actions seem to be _toggling_ and our action creator is even named `toggleNoteInfo` we can probably simplify this by arranging our state into a toggle paradigm and skip the associated boolean operator.

```
const showNoteInfo: A.Reducer<boolean> = (state = false, action) => {
  switch (action.type) {
    case 'NOTE_INFO_TOGGLE':
      return !state;

    case 'App.toggleNavigation': // or whatever
      return false;

    default:
      return state;
  }
}
```

It's in this manner that we can move from asking "what data does this change?" as seen in `app-state.ts` to "what changes this data?" where the reducers only need to govern their own type. if we run into problems with things like the fact that `toggleNavigation` might turn _on_ the note info panel then we just need to be creative, but as in many places so in this PR, I think that it's safe to say that `toggleNavigation` should _always_ disable the note info panel because _either_ it's disabling it while opening navigation _or_ it would have already been disabled when closing navigation

Reply:
Yeah, I thought about that. Do we need to worry about race conditions (which is usually the only reason to be explicit about passing something to a toggle)?

It's also sort of nice that it's very explicit -- the "open" action passes `true` and the "close" action passes `false` -- but not really necessary.

Reply:
> Do we need to worry about race conditions

The nice thing about simple reducers vs. imperative updates or _Redux thunks_ is that there are no race conditions. Each reducer updates its own data only in response to the actions initiated by the user _or_ by some network activity or timer event.

For the race to occur we'd need to be able to dispatch and update state after _some_ reducers have updated but before _this_ reducer updates, which isn't possible.